datadog/resource_datadog_metric_tag_configuration.go

```diff
@@ -161,20 +173,26 @@ func buildDatadogMetricTagConfiguration(d *schema.ResourceData) (*datadogV2.Metr
 	}
 	attributes.SetMetricType(*metricType)
 
-	includePercentiles, iclFieldSet := d.GetOk("include_percentiles")
+	includePercentiles, _ := d.GetOk("include_percentiles")
 
-	if iclFieldSet {
-		if *metricType != datadogV2.METRICTAGCONFIGURATIONMETRICTYPES_DISTRIBUTION {
-			return nil, fmt.Errorf("include_percentiles field not allowed with metric_type: %s, only with metric_type distribution", *metricType)
-		}
-		attributes.SetIncludePercentiles(includePercentiles.(bool))
+	if includePercentiles.(bool) && *metricType != datadogV2.METRICTAGCONFIGURATIONMETRICTYPES_DISTRIBUTION {
+		return nil, fmt.Errorf("include_percentiles field not allowed with metric_type: %s, only with metric_type distribution", *metricType)
+	}
+
+	if *metricType != datadogV2.METRICTAGCONFIGURATIONMETRICTYPES_DISTRIBUTION {
+		attributes.IncludePercentiles = nil
```

Comment:
Potentially a breaking change; please see updated PR description for more details

Reply:
I don't think this breaking change is actually a problem - the only case I can find where this differs in behavior to the old version is when `include_percentiles = true` on the backend, but is omitted or false in the terraform config. In this case, the old version actually has perpetual drift (terraform keeps trying to set it to null/false, but the provider doesn't actually send a false to backend), and I think its very unlikely a user would leave their config in this state.